spec.html

```diff
@@ -25659,29 +25659,27 @@ <h1>Runtime Semantics: Decode ( _string_, _reservedSet_ )</h1>
                 1. If the code units at index (_k_ + 1) and (_k_ + 2) within _string_ do not represent hexadecimal digits, throw a *URIError* exception.
                 1. Let _B_ be the 8-bit value represented by the two hexadecimal digits at index (_k_ + 1) and (_k_ + 2).
                 1. Set _k_ to _k_ + 2.
-                1. If the most significant bit in _B_ is 0, then
+                1. Let _n_ be the number of leading 1 bits in _B_.
+                1. If _n_ = 0, then
                   1. Let _C_ be the code unit whose value is _B_.
                   1. If _C_ is not in _reservedSet_, then
                     1. Let _S_ be the String value containing only the code unit _C_.
                   1. Else,
                     1. Let _S_ be the substring of _string_ from _start_ to _k_ + 1.
                 1. Else,
-                  1. Assert: the most significant bit in _B_ is 1.
-                  1. Let _n_ be the smallest nonnegative integer such that (_B_ &lt;&lt; _n_) &amp; 0x80 is equal to 0.
-                  1. If _n_ equals 1 or _n_ is greater than 4, throw a *URIError* exception.
-                  1. Let _Octets_ be a List of 8-bit integers of size _n_.
-                  1. Set _Octets_[0] to _B_.
+                  1. If _n_ = 1 or _n_ &gt; 4, throw a *URIError* exception.
                   1. If _k_ + (3 &times; (_n_ - 1)) is greater than or equal to _strLen_, throw a *URIError* exception.
+                  1. Let _Octets_ be a new List containing _B_.
```

Comment:
```suggestion
                  1. Let _Octets_ be &laquo; _B_ &raquo;.
```
but it's fine either way